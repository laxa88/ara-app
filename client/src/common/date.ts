// https://stackoverflow.com/questions/23593052/format-javascript-date-to-yyyy-mm-dd

const getDateData = () => {
  const d = new Date();

  const year = d.getFullYear();
  let month = `${d.getMonth() + 1}`;
  let day = `${d.getDate()}`;

  if (month.length < 2) {
    month = `0${month}`;
  }
  if (day.length < 2) {
    day = `0${day}`;
  }

  return { year, month, day };
};

/**
 * Returns current month, e.g. "2018-11"
 */
const getMonth = () => {
  const { year, month } = getDateData();
  const resultDate = [year, month].join('-');
  return resultDate;
};

/**
 * Returns today's date, e.g. "2018-11-16"
 */
const getDate = () => {
  const { year, month, day } = getDateData();
  const resultDate = [year, month, day].join('-');
  return resultDate;
};

/**
 * Returns current time, e.g. "23:42:09"
 */
const getTime = () => {
  const d = new Date();

  let h = d.getHours().toString();
  let m = d.getMinutes().toString();
  let s = d.getSeconds().toString();

  if (h.length < 2) {
    h = `0${h}`;
  }

  if (m.length < 2) {
    m = `0${m}`;
  }

  if (s.length < 2) {
    s = `0${s}`;
  }

  const resultTime = [h, m, s].join(':');

  return resultTime;
};

/**
 * Returns formatted date, e.g. "2018-11-16 23:42:09"
 */
const formatDate = () => {
  const result = `${getDate()} ${getTime()}`;
  return result;
};

/**
 * Parses data, e.g. "2018-12-01T15:00:00.000Z" to "2018-12-01"
 */
const parseDate = (date: string) => {
  return date.substr(0, 10);
};

export { getMonth, getDate, getTime, formatDate, parseDate };
