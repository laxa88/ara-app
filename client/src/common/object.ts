const getValue = (object: any, propName: string) => {
  const arr = propName.split('.');
  let result = object;

  while (arr.length && result) {
    result = result[arr.shift()];
  }

  return result;
};

export { getValue };
