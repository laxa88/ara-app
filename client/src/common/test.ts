import { configure, mount, render, shallow } from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import reduxMockStore from 'redux-mock-store';

configure({ adapter: new Adapter() });

// Reserved for later on, if we use thunk or saga or anything else
const middleware: any = [];

function mockStore(state: any) {
  return reduxMockStore(middleware)(state);
}

export { shallow, mount, render, mockStore };
