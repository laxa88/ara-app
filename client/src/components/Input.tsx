import * as React from 'react';
import styled from '../css/styled';

const StyledInput = styled.input`
  font-size: 1em;
  margin: 1em;
  padding: 0.25em 1em;
  border-radius: 3px;
`;

export interface IInput {
  value: string;
  placeholder?: string;
  type?: string;
  onChange: (newValue: string, prevValue: string) => void;
}

class Input extends React.PureComponent<IInput, {}> {
  constructor(props: IInput) {
    super(props);

    this.onChange = this.onChange.bind(this);
  }

  public render() {
    const { value, placeholder, onChange, ...others } = this.props;

    return (
      <div>
        <StyledInput
          value={value}
          placeholder={placeholder}
          onChange={this.onChange}
          {...others}
        />
      </div>
    );
  }

  private onChange(e: React.ChangeEvent<HTMLInputElement>) {
    const { value: newValue } = e.currentTarget;
    const { value: prevValue } = this.props;
    this.props.onChange(newValue, prevValue);
  }
}

export default Input;
