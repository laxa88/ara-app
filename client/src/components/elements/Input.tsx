import * as React from 'react';
import styled from '../css/styled';

interface IInput {
  isActive: boolean;
}

interface IInput2 {
  primary: boolean;
}

const Input = styled<IInput, 'input'>('input')`
  font-size: 1em;
  margin: 1em;
  padding: 0.25em 1em;
  border-radius: 3px;

  /* Color the border and text with theme.main */
  color: ${props => props.theme.primaryColor};
  border: 2px solid
    ${props =>
      props.isActive
        ? props.theme.primaryColor
        : props.theme.primaryColorInverted};
`;

const Input2 = styled<IInput2, 'input'>('input')`
  font-size: 1em;
  margin: 1em;
  padding: 0.25em 1em;
  border-radius: 3px;

  /* Color the border and text with theme.main */
  color: ${props => props.theme.primaryColorInverted};
  border: 2px solid ${props => (props.primary ? 'blue' : 'yellow')};
`;

Input.defaultProps = {
  theme: {
    primaryColor: 'palevioletred',
    primaryColorInverted: 'green',
  },
};

Input2.defaultProps = {
  theme: {
    primaryColor: 'palevioletred',
    primaryColorInverted: 'green',
  },
};

interface IProps {
  isActive: boolean;
  placeholder: string;
  type?: string;
}

interface IState {
  value: string;
}

class InputBase extends React.Component<IProps, IState> {
  constructor(props: IProps) {
    super(props);

    this.onChange = this.onChange.bind(this);

    this.state = {
      value: '',
    };
  }

  public onChange(e: React.ChangeEvent<HTMLInputElement>) {
    this.setState({ value: e.currentTarget.value });
  }

  public render() {
    const { placeholder, ...others } = this.props;

    return (
      <div>
        <Input
          placeholder={placeholder}
          value={this.state.value}
          onChange={this.onChange}
          {...others}
        />
        <Input2 primary={this.props.isActive} />
      </div>
    );
  }
}

export default InputBase;
