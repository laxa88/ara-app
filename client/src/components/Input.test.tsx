import * as React from 'react';
import { mount, shallow } from '../common/test';
import Input, { IInput } from './Input';

describe('<Input />', () => {
  const mockProps = {
    onChange: jest.fn(),
    value: 'dummy value',
  };

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should render', () => {
    const wrapper = shallow(<Input {...mockProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('onChange() should trigger', () => {
    // Arrange

    const mockEvent = {
      currentTarget: { value: 'dummy new value' },
    } as React.ChangeEvent<HTMLInputElement>;

    const wrapper = mount(<Input {...mockProps} />);

    const inputProps: React.HTMLAttributes<HTMLInputElement> = wrapper
      .find('input')
      .props();

    // Act

    inputProps.onChange(mockEvent);

    // Assert

    expect(mockProps.onChange).toHaveBeenCalledWith(
      mockEvent.currentTarget.value,
      mockProps.value,
    );
  });
});
