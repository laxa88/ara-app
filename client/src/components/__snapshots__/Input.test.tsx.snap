// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Input
    onChange={[MockFunction]}
    value="dummy value"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ForwardRef
        onChange={[Function]}
        value="dummy value"
      />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "onChange": [Function],
        "placeholder": undefined,
        "value": "dummy value",
      },
      "ref": null,
      "rendered": null,
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": undefined,
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
  font-size: 1em;
  margin: 1em;
  padding: 0.25em 1em;
  border-radius: 3px;
",
          ],
        },
        "displayName": "styled.input",
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "input",
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ForwardRef
          onChange={[Function]}
          value="dummy value"
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "onChange": [Function],
          "placeholder": undefined,
          "value": "dummy value",
        },
        "ref": null,
        "rendered": null,
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "rules": Array [
              "
  font-size: 1em;
  margin: 1em;
  padding: 0.25em 1em;
  border-radius: 3px;
",
            ],
          },
          "displayName": "styled.input",
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "input",
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
